test/unit/model_test.rb

```diff
@@ -978,100 +978,94 @@ def without(&block)
+      should 'version.object_changes attribute should have stored the value returned by the attribute serializer' do
+        as_stored_in_version = HashWithIndifferentAccess[YAML::load(@person.versions.last.object_changes)]
+        assert_equal [nil, 'Samoa'], as_stored_in_version[:time_zone]
+        serialized_value = Person::TimeZoneSerializer.dump(@person.time_zone)
+        assert_equal serialized_value, as_stored_in_version[:time_zone].last
+      end
 
-      should "be an instance of ActiveSupport::TimeZone" do
-        assert_equal ActiveSupport::TimeZone, @person.time_zone.class
+      # Tests for unserialization:
+      should 'version.changeset should convert the attribute value back to its original, unserialized value' do
+        unserialized_value = Person::TimeZoneSerializer.load(@person.time_zone)
+        assert_equal unserialized_value, @person.versions.last.changeset[:time_zone].last
+      end
+      should "record.changes (before save) returns the original, unserialized values" do
+        assert_equal [NilClass, ActiveSupport::TimeZone], @changes_before_save[:time_zone].map(&:class)
+      end
+      should 'version.changeset should be the same as record.changes was before the save' do
+        assert_equal @changes_before_save, @person.versions.last.changeset.delete_if { |key, val| key.to_sym == :id }
+        assert_equal [NilClass, ActiveSupport::TimeZone], @person.versions.last.changeset[:time_zone].map(&:class)
       end
 
-      context 'when the model is saved' do
+      context 'when that attribute is updated' do
         setup do
+          @attribute_value_before_change = @person.time_zone
+          @person.assign_attributes({ :time_zone => 'Pacific Time (US & Canada)' })
           @changes_before_save = @person.changes.dup
           @person.save!
         end
 
-        # Test for serialization:
+        # Tests for serialization:
+        # Before the serialized attributes fix, the object/object_changes value that was stored was ridiculously long (58723).
+        should 'version.object should not have stored the default, ridiculously long (to_yaml) serialization of the TimeZone object' do
+          assert @person.versions.last.object.length < 105, "object length was #{@person.versions.last.object.length}"
+        end
+        # Need an additional clause to detect what version of ActiveRecord is being used for this test because AR4 injects the `updated_at` column into the changeset for updates to models
         should 'version.object_changes should not have stored the default, ridiculously long (to_yaml) serialization of the TimeZone object' do
-          assert @person.versions.last.object_changes.length < 105, "object_changes length was #{@person.versions.last.object_changes.length}"
+          assert @person.versions.last.object_changes.length < (ActiveRecord::VERSION::MAJOR < 4 ? 105 : 118), "object_changes length was #{@person.versions.last.object_changes.length}"
```

Comment:
You sure, you want me to make this change? It looks the same on master: https://github.com/airblade/paper_trail/blob/94db8436b58b476963110e99926588b621fe72a0/test/unit/model_test.rb#L1041.

All I've done is outdent by two spaces, because I deleted the surrounding Active Record version check. Try appending `?w=1` to the files URL, then GitHub won't show whitespace changes. :grin: